### PR TITLE
dep: move to Go modules

### PR DESCRIPTION
**Pull request by samuelkarp**
cc @nmeyerhans @sipsma @xibz @aaithal @jmillikin-stripe

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,33 +24,21 @@ PUSH_BINARY=$(ROOT)/bin/ecr-push
...
-$(PULL_BINARY): $(SOURCES) $(VENDORDIR)
-	cd $(PULLDIR) && go build -o $(PULL_BINARY) .
-
-$(PUSH_BINARY): $(SOURCES) $(VENDORDIR)
-	cd $(PUSHDIR) && go build -o $(PUSH_BINARY) .
-
-$(COPY_BINARY): $(SOURCES) $(VENDORDIR)
-	cd $(COPYDIR) && go build -o $(COPY_BINARY) .
+$(PULL_BINARY): $(SOURCES)
+	cd $(PULLDIR) && GO111MODULE=on go build -o $(PULL_BINARY) .
```
Comment (nmeyerhans) [inline Makefile]: Should we `export GO111MODULE=on` early in the Makefile, such that it's applied globally, instead of repeating it everywhere?

**Comment by xibz** [comment]
Have you ran `go mod tidy`? Running it locally removed some unneeded packages.

<details>
<pre>
 require (
-       github.com/Microsoft/go-winio v0.4.12
-       github.com/Microsoft/hcsshim v0.8.6
+       github.com/Microsoft/go-winio v0.4.12 // indirect
+       github.com/Microsoft/hcsshim v0.8.6 // indirect
        github.com/aws/aws-sdk-go v0.0.0-20190610182410-b348cb633fa4
-       github.com/containerd/console v0.0.0-20181022165439-0650fd9eeb50
+       github.com/containerd/cgroups v0.0.0-20190717030353-c4b9ac5c7601 // indirect
+       github.com/containerd/console v0.0.0-20181022165439-0650fd9eeb50 // indirect
        github.com/containerd/containerd v1.2.7
-       github.com/containerd/continuity v0.0.0-20190426062206-aaeac12a7ffc
-       github.com/containerd/fifo v0.0.0-20190226154929-a9fb20d87448
-       github.com/containerd/typeurl v0.0.0-20190515163108-7312978f2987
-       github.com/davecgh/go-spew v1.1.1
-       github.com/docker/distribution v0.0.0-20190205005809-0d3efadf0154
+       github.com/containerd/continuity v0.0.0-20190426062206-aaeac12a7ffc // indirect
+       github.com/containerd/fifo v0.0.0-20190226154929-a9fb20d87448 // indirect
+       github.com/containerd/typeurl v0.0.0-20190515163108-7312978f2987 // indirect
+       github.com/coreos/go-systemd v0.0.0-20190719114852-fd7a80b32e1f // indirect
+       github.com/docker/distribution v0.0.0-20190205005809-0d3efadf0154 // indirect
        github.com/docker/docker v0.0.0-20190607191414-238f8eaa31aa
-       github.com/docker/go-events v0.0.0-20170721190031-9461782956ad
+       github.com/docker/go-events v0.0.0-20170721190031-9461782956ad // indirect
        github.com/docker/go-units v0.4.0
-       github.com/gogo/googleapis v1.2.0
-       github.com/gogo/protobuf v1.2.1
-       github.com/golang/protobuf v1.3.1
-       github.com/jmespath/go-jmespath v0.0.0-20180206201540-c2b33e8439af
-       github.com/konsorten/go-windows-terminal-sequences v1.0.2
+       github.com/godbus/dbus v4.1.0+incompatible // indirect
+       github.com/gogo/googleapis v1.2.0 // indirect
+       github.com/konsorten/go-windows-terminal-sequences v1.0.2 // indirect
        github.com/opencontainers/go-digest v0.0.0-20190228220655-ac19fd6e7483
        github.com/opencontainers/image-spec v0.0.0-20190321123305-da296dcb1e47
-       github.com/opencontainers/runc v0.1.1
-       github.com/opencontainers/runtime-spec v1.0.1
+       github.com/opencontainers/runc v0.1.1 // indirect
+       github.com/opencontainers/runtime-spec v1.0.1 // indirect
        github.com/pkg/errors v0.0.0-20190227000051-27936f6d90f9
-       github.com/pmezard/go-difflib v1.0.0
-       github.com/sirupsen/logrus v1.4.2
+       github.com/sirupsen/logrus v1.4.2 // indirect
        github.com/stretchr/testify v1.2.2
-       github.com/syndtr/gocapability v0.0.0-20180916011248-d98352740cb2
+       github.com/syndtr/gocapability v0.0.0-20180916011248-d98352740cb2 // indirect
        golang.org/x/net v0.0.0-20190607181551-461777fb6f67
        golang.org/x/sync v0.0.0-20190423024810-112230192c58
-       golang.org/x/sys v0.0.0-20190610200419-93c9922d18ae
-       golang.org/x/text v0.3.2
-       google.golang.org/genproto v0.0.0-20190605220351-eb0b1bdb6ae6
-       google.golang.org/grpc v1.21.1
+       golang.org/x/sys v0.0.0-20190610200419-93c9922d18ae // indirect
+       golang.org/x/text v0.3.2 // indirect
+       google.golang.org/genproto v0.0.0-20190605220351-eb0b1bdb6ae6 // indirect
+       google.golang.org/grpc v1.21.1 // indirect
+       gotest.tools v2.2.0+incompatible // indirect
 )
</pre>
</details>

File: .travis.yml [hunk 2]
```diff
@@ -3,9 +3,8 @@ sudo: false
 os:
     - linux
 go:
-    - 1.10.x
+    - 1.12.x
```
Comment (sipsma) [inline .travis.yml]: nit: is 1.12 a hard requirement? or is it just 1.11 and above? If it's just 1.11 and above you can specify both versions here and have travis test both

File: Makefile (hunk 1 above)
Comment (samuelkarp) [reply Makefile]: Done

File: .travis.yml (hunk 2 above)
Comment (samuelkarp) [reply .travis.yml]: Done

**Comment by samuelkarp** [comment]
@xibz Done, thanks!

**Pull request merged**